10_construct-database/13_construct_county_race-share-geography.qmd

```diff
@@ -0,0 +1,259 @@
+  write_csv(
+    here("data", "temp", "transit_cost_all_county_subgroups_race_share.csv")
+  )
+
+```
+
+### Racial share files
+
+Create a population file for the race-share subgroup. 
+
+```{r}
+
+read_csv(here("geographic-crosswalks", "data", "county-populations.csv")) %>% 
+  mutate(all = "All", 
+         race_non_white = "Majority Non-White", 
+         race_white_non_hisp = "Majority White, Non-Hispanic",
+         race_mixed = "Mixed Race and Ethnicity") %>% 
+  pivot_longer(cols = c(all:race_mixed),
+               names_to = "subgroup_type",
+               values_to = "subgroup") %>% 
+  mutate(subgroup_type = ifelse(subgroup_type != "all", "race-ethnicity", "all")) %>% 
+  write_csv(here("data", "temp", "county-populations_race-share.csv"))
+
+```
+
+Read file paths to all county files with race-share information. 
+
+```{r}
+
+filepaths <- c(
+
+# Race-share populations
+here("data", "temp", "county-populations_race-share.csv"),
+
+# 01 financial well-being
+here("data", "temp", "county_debt_longitudinal_subgroup_race_share.csv"),
+
+# 06 neighborhoods
+here("data", "temp", "environment_county_subroup_race_share.csv"),
+here("data", "temp", "transit_trips_all_county_subgroups_race_share.csv"),
+here("data", "temp", "transit_cost_all_county_subgroups_race_share.csv")
+
+)
+
+```
+
+This code loads each file and then combines them using `left_join()`. The first file is a population file, so there should be 3,143 counties per year. 
+
+```{r}
+
+db_race_share <- filepaths |>
+  map(
+    .f = ~read_csv(.x) |> 
+      select(-any_of(c("state_name", "county_name")))
+  ) |>
+  reduce(left_join, by = c("year", "state", "county", "subgroup", "subgroup_type"))
+
+```
+
+## Quality and Completeness
+
+### Dimensions
+
+At most there should be 12,576 in a year (4X3144) and at the least there should be 12,568 in a year (4X3142).
+
+```{r}
+dimension_test <- function(.data) {
+  .data |> 
+  count(year) |>
+  assert(
+    within_bounds(12568, 22008), 
+    n
+  )
+}
+
+dimension_test(db_race_share)
+
+```
+
+### Data Quality Flags
+
+This section summarizes data quality flags for variables and years.
+
+```{r}
+db_race_share |>
+  select(ends_with("_quality")) |>
+  pivot_longer(everything(), names_to = "variable", values_to = "quality") |>
+  count(variable, quality) |>
+  filter(!is.na(quality)) |>
+  print(n = 100)
+
+db_race_share  |>
+  select(year, ends_with("_quality")) |>
+  pivot_longer(-year, names_to = "variable", values_to = "quality") |>
+  count(year, quality) |>
+  print(n = 100)
+
+```
+
+### Full Database
+
+Evaluate the quality variables. 
+
+```{r}
+db_race_share |>
+  select(ends_with("_quality")) |>
+  assert(in_set(1, 2, 3), everything())
+
+```
+
+
+### 01 Financial Well-Being
+
+```{r}
+
+db_race_share |>
+  assert(
+    within_bounds(0, Inf), 
+    share_debt_coll, 
+    share_debt_coll_ub, 
+    share_debt_coll_lb
+  ) |>
+  quiet()
```

Comment:
Does this not work for `within_bounds(0, 1)`?

Same question for the overall file (which would 100% be my fault!).